### PR TITLE
docs(API): add picture endpoints in API specs #BLOG-39

### PR DESCRIPTION
**Pull request by fenohasinalala**

### CODE REVIEW

File: doc/api.yml [hunk 1]
```diff
@@ -958,6 +1237,19 @@ components:
             $ref: "#/components/schemas/Category"
         is_followed:
           type: boolean
+    UserPicture:
+      properties:
+        user:
+          $ref: "#/components/schemas/User"
```
Comment (YumeT023) [inline doc/api.yml]: We may not need the entire user info, user_id is enough

File: doc/api.yml [hunk 2]
```diff
@@ -877,6 +1145,17 @@ components:
             $ref: "#/components/schemas/Category"
         reactions:
           $ref: "#/components/schemas/ReactionStat"
+
+    PostPicture:
+      properties:
+        id:
+          type: string
+        post:
```
Comment (YumeT023) [inline doc/api.yml]: We don't need the entire post
post_id is enough

File: doc/api.yml [hunk 3]
```diff
@@ -877,6 +1145,17 @@ components:
...
+          $ref: "#/components/schemas/Post"
+        document_reference:
```
Comment (YumeT023) [inline doc/api.yml]: Is this used in the frontend?

Are these informations really necessary?

I think the only thing that is going to be used is the picture url
> Unless there is a specific use-case you want to address here

File: doc/api.yml [hunk 4]
```diff
@@ -958,6 +1237,19 @@ components:
...
+          $ref: "#/components/schemas/User"
+        type:
+          $ref: "#/components/schemas/UserPicture"
+        url:
+          type: string
+    UserPictureType:
+        type: string
+        enum:
+         - BANNER
+         - PHOTO
```
Comment (YumeT023) [inline doc/api.yml]: > PROFILE

File: doc/api.yml [hunk 5]
```diff
@@ -577,6 +694,157 @@ paths:
...
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
```
Comment (YumeT023) [inline doc/api.yml:844]: rm unused spaces

File: doc/api.yml [hunk 6]
```diff
@@ -577,6 +694,157 @@ paths:
...
+          schema:
+            type: string
+        - name: picId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Post picture deleted.
```
Comment (YumeT023) [inline doc/api.yml]: The deleted picture.

File: doc/api.yml [hunk 7]
```diff
@@ -577,6 +694,157 @@ paths:
...
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  "/posts/{pId}/pictures/{picId}":
+    post:
+      tags:
+        - Posting
+      summary: Add one picture to a post
```
Comment (YumeT023) [inline doc/api.yml]: Upload a picture to a post.

File: doc/api.yml [hunk 8]
```diff
@@ -577,6 +694,157 @@ paths:
...
+      operationId: postPostPicture
```
Comment (YumeT023) [inline doc/api.yml]: uploadPostPicture

File: doc/api.yml [hunk 9]
```diff
@@ -577,6 +694,157 @@ paths:
...
+          schema:
+            type: string
+        - name: picId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Post pictures fetched.
```
Comment (YumeT023) [inline doc/api.yml:716]: The identified post pictures.

File: doc/api.yml [hunk 10]
```diff
@@ -222,6 +222,123 @@ paths:
...
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
```
Comment (YumeT023) [inline doc/api.yml:339]: unused spaces

File: doc/api.yml [hunk 11]
```diff
@@ -577,6 +694,157 @@ paths:
...
+          schema:
+            type: string
+        - name: picId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Post picture fetched.
```
Comment (YumeT023) [inline doc/api.yml:797]: The identified picture.

File: doc/api.yml [hunk 12]
```diff
@@ -222,6 +222,123 @@ paths:
...
                 $ref: "#/components/schemas/User"
 
+  "/users/{uId}/pictures":
+    put:
+      tags:
+        - User
+      summary: Submit user picture of the type profile banner or profile picture
+      operationId: putUserPicture
+      parameters:
+        - name: uId
```
Comment (YumeT023) [inline doc/api.yml]: uid , no uppercase

**Review by YumeT023 — CHANGES_REQUESTED** [review]

File: doc/api.yml (hunk 3 above)
Comment (fenohasinalala) [reply doc/api.yml]: document_reference serves as a placeholder that the front puts in the content (document) when they create/update a Post, so we can replace them later when someone requests a Post. 
I'll change the name to "placeholder"

Comment (YumeT023) [reply doc/api.yml]: Just to be certain, they're replaced (mapped) when someone requests the associated Post? we don't need to replace them manually in the frontend ?

File: doc/api.yml [hunk 13]
```diff
@@ -222,6 +222,121 @@ paths:
               schema:
                 $ref: "#/components/schemas/User"
 
+  "/users/{uid}/pictures":
```
Comment (YumeT023) [inline doc/api.yml:225]: I think that this endpoint can only be replace by the associated user (similar to getSelfFollowers) but more flexible in this way, make sure it is thoroughly tested in your implementation (security)

**Review by YumeT023 — CHANGES_REQUESTED** [review]

File: doc/api.yml [hunk 14]
```diff
@@ -958,6 +1234,19 @@ components:
             $ref: "#/components/schemas/Category"
         is_followed:
           type: boolean
+    UserPicture:
+      properties:
+        user_id:
+          type: string
+        type:
+          $ref: "#/components/schemas/UserPicture"
```
Comment (YumeT023) [inline doc/api.yml]: wrong $ref
UserPictureType instead

**Review by YumeT023 — CHANGES_REQUESTED** [review]

File: doc/api.yml (hunk 13 above)
Comment (fenohasinalala) [reply doc/api.yml:225]: yes

**Pull request merged**